src/sentry/models/group.py

```diff
@@ -118,6 +118,31 @@ def from_kwargs(self, project, **kwargs):
                 }
             )
 
+    def from_event_id(self, project, event_id):
+        """
+        Resolves the 32 character event_id string into
+        a Group for which it is found.
+        """
+        from sentry import features
+
+        if features.has('projects:sample-events', project=self):
+            from sentry.models import EventMapping as Model
+        else:
+            from sentry.models import Event as Model
+
+        try:
+            group_id = Model.objects.filter(
+                project_id=self.id,
+                event_id=event_id,
+            ).values_list('group_id', flat=True)[0]
```

Comment:
Can you do a `select_related('group')` here instead of loading the group separately?

Reply:
Group is not actually foreign key, and it's on a separately physical database.